server/canvas_api_response.rb

```diff
@@ -14,15 +14,13 @@ class CanvasAPIResponse
   # Store the body of the response.  All output from Canvas API goes through this class.
   attr_accessor :body_string, :body_json
 
-  ## TODO: allow input as string or json
-
-  # Input is in string format
-  def initialize(body)
+  # Input is in string format.  Some values (e.g. host names) may need to be changed.
+  def initialize(body,replace)
```

Comment:
I'd like to see a more descriptive name for "replace".  For example, "stringReplacementTable".